ProjFS.Mac/PrjFSKextTests/KauthHandlerTests.mm

```diff
@@ -58,4 +81,179 @@ - (void)testShouldIgnoreVnodeType {
+                                            &vnodeFileFlags,
+                                            &pid,
+                                            procname,
+                                            &kauthResult,
+                                            &kauthError));
+    XCTAssertEqual(kauthResult, KAUTH_RESULT_DEFER);
+
+    
+    // Test failure reading attr
+    testVnode->errors.getattr = EBADF;
+    XCTAssertFalse(ShouldHandleVnodeOpEvent(
+                                            &perfTracer,
+                                            context,
+                                            testVnode.get(),
+                                            action,
+                                            &vnodeType,
+                                            &vnodeFileFlags,
+                                            &pid,
+                                            procname,
+                                            &kauthResult,
+                                            &kauthError));
+    XCTAssertEqual(kauthResult, KAUTH_RESULT_DENY);
+    // reset to valid value
+    testVnode->errors.getattr = 0;
+
+    
+    // Test invalid file flag
+    testVnode->values.getattr = FileFlags_IsEmpty;
+    XCTAssertFalse(ShouldHandleVnodeOpEvent(
+                                            &perfTracer,
+                                            context,
+                                            testVnode.get(),
+                                            action,
+                                            &vnodeType,
+                                            &vnodeFileFlags,
+                                            &pid,
+                                            procname,
+                                            &kauthResult,
+                                            &kauthError));
+    XCTAssertEqual(kauthResult, KAUTH_RESULT_DEFER);
+    // reset to valid value
+    testVnode->values.getattr = FileFlags_IsInVirtualizationRoot;
+    
+
+    // Test with file crawler trying to populate an empty file
+    testVnode->values.getattr = FileFlags_IsEmpty | FileFlags_IsInVirtualizationRoot;
+    strcpy(procname, "mds");
+    XCTAssertFalse(ShouldHandleVnodeOpEvent(
+                                            &perfTracer,
+                                            context,
+                                            testVnode.get(),
+                                            action,
+                                            &vnodeType,
+                                            &vnodeFileFlags,
+                                            &pid,
+                                            procname,
+                                            &kauthResult,
+                                            &kauthError));
+    XCTAssertEqual(kauthResult, KAUTH_RESULT_DENY);
+
+    
+    // Test with finder trying to populate an empty file
+    strcpy(procname, "finder");
```

Comment:
Nit: The `"Finder"` process is capitalised, but it shouldn't matter for the purposes of the test.